libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2278,27 +2316,32 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			auto const& toStructType = dynamic_cast<StructType const &>(_to);
 			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
 
-			solUnimplementedAssert(!fromStructType.isDynamicallyEncoded(), "");
-			solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
-			solUnimplementedAssert(fromStructType.location() != DataLocation::Memory, "");
-
-			if (fromStructType.location() == DataLocation::CallData)
+			if (fromStructType.location() == toStructType.location() && toStructType.isPointer())
 			{
-				body = Whiskers(R"(
-					converted := <abiDecode>(value, calldatasize())
-				)")("abiDecode", ABIFunctions(m_evmVersion, m_revertStrings, m_functionCollector).tupleDecoder(
-					{&toStructType}
-				)).render();
+				solUnimplementedAssert(toStructType.sizeOnStack() == 1 && fromStructType.sizeOnStack() == 1, "");
```

Comment:
Just because we were messing with this simultanously, so maybe otherwise the comment will get lost ;-):
I think this rather belongs in L2184 right when the function signature is first created.

Reply:
(Then based on ``_from.sizeOnStack()`` and ``_to.sizeOnStack()`` - but that's the place that will really be hit, if they're not both one)

Reply:
I added it. But does it really makes sense, as there is a call before this:
```
if (_from.sizeOnStack() != 1 || _to.sizeOnStack() != 1)
	return conversionFunctionSpecial(_from, _to);
```

maybe `solAssert` instead?

Reply:
Ah, I didn't see that - that's a rather unfortunate distinction and scatters the array conversions all over the place... we should think about changing this in the future, but not in this PR :-).
But we can just drop the assertion in this case, it was overdoing it anyways.

Reply:
> Ah, I didn't see that - that's a rather unfortunate distinction and scatters the array conversions all over the place... we should think about changing this in the future, but not in this PR :-).
> But we can just drop the assertion in this case, it was overdoing it anyways.

Yes, it looks very unfortunate...
Once I finish copying of arrays to storage will maybe take a look at this and do the refactor.